src/System.IO.FileSystem.Watcher/tests/FileSystemWatcher.LockTest.Unix.cs

```diff
@@ -0,0 +1,28 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO;
+using Xunit;
+
+public class FileSystemWatcherLockTest
+{
+    [Fact]
+    public static void FileSystemWatcher_WatchLocksWatchedDirectory()
+    {
+        using (var dir = Utility.CreateTestDirectory())
+        using (var fsw = new FileSystemWatcher())
+        {
+            fsw.Path = Path.GetFullPath(dir.Path);
+            fsw.EnableRaisingEvents = true;
+
+            Microsoft.Win32.SafeHandles.SafeFileHandle handle = Interop.Sys.Open(fsw.Path, Interop.Sys.OpenFlags.O_RDONLY, 0);
+            Assert.False(handle.IsInvalid); // We should always be able to open a handle
+            Assert.True(Interop.Sys.FLock(handle, Interop.Sys.LockOperations.LOCK_EX | Interop.Sys.LockOperations.LOCK_NB) < 0);
+
+            Interop.ErrorInfo err = Interop.Sys.GetLastErrorInfo();
+            Assert.True(err.Error == Interop.Error.EWOULDBLOCK); // validate it is a sharing violation
+        }
+    }
+}
```

Comment:
So the overall goal of this PR was to add an advisory lock on the root folder to let other apps know that we were using this folder to be good citizens, since Windows has a shared lock. Since Linux has a different type of lock mechanism that doesn't correlate with Windows, I'm cool dropping this pack and closing the bug if you don't think it's necessary :)